code/obj/decorations.dm

```diff
@@ -746,6 +751,151 @@
 	name = "W blind switch"
 	pixel_x = -24
 
+/obj/sign_switch
+	name = "sign switch"
+	desc = "Connected to one or more illuminated signs, turning them on or off."
+	icon = 'icons/obj/power.dmi'
+	icon_state = "light0"
+	anchored = 1
+	density = 0
+	var/on = FALSE
+	var/id = null
+
+	New()
+		..()
+		if (current_state > GAME_STATE_PREGAME)
+			SPAWN(0.5 SECONDS)
+				src.initialize()
+
+	initialize()
+		if (!src.name || (src.name in list("N sign switch", "E sign switch", "S sign switch", "W sign switch")))
+			src.name = "sign switch"
+		if (!src.id)
+			var/area/sign_area = get_area(src)
+			src.id = sign_area.name
+		ADD_SWITCHED_OBJ(SWOB_SIGNAGE)
+
+	disposing()
+		REMOVE_SWITCHED_OBJ(SWOB_SIGNAGE)
+		. = ..()
+
+	ex_act(var/severity)
+		switch(severity)
+			if(1,2)
+				qdel(src)
+			else
+				if(prob(50))
+					qdel(src)
+
+	proc/toggle(var/new_state)
+		src.on = new_state
+		src.icon_state = "light[src.on]"
+		src.UpdateIcon()
+
+	proc/toggle_group()
+		if(!ON_COOLDOWN(src, "toggle", 1 SECOND))
+			switched_obj_toggle(SWOB_SIGNAGE,src.id,!(src.on))
+			playsound(src, 'sound/misc/lightswitch.ogg', 50, 1)
+
+	attack_hand(mob/user)
+		src.toggle_group()
+
+	attack_ai(mob/user as mob)
+		src.toggle_group()
+
+	attackby(obj/item/W, mob/user)
+		src.toggle_group()
+
+/obj/sign_switch/north
+	name = "N sign switch"
+	pixel_y = 24
+
+/obj/sign_switch/east
+	name = "E sign switch"
+	pixel_x = 24
+
+/obj/sign_switch/south
+	name = "S sign switch"
+	pixel_y = -24
+
+/obj/sign_switch/west
+	name = "W sign switch"
+	pixel_x = -24
+
+/obj/machinery/illuminated_sign
+	name = "illuminated sign"
+	desc = "It's a sign on the wall that does the glowy thing."
+	icon = 'icons/obj/decoration.dmi'
+	icon_state = "occupancy-1"
+	anchored = 1
+	density = 0
+	opacity = 0
+	layer = FLY_LAYER+1.01 // just above windows
```

Comment:
why not use the window layer then?

Reply:
This persists from the old implementation. If WINDOW_LAYER+0.01 would be preferable, I can standardize to that in this context.

Reply:
yes

Reply:
WINDOW_LAYER.... apparently doesn't actually exist. And the blinds show up underneath the windows when I use COG2_WINDOW_LAYER (the closest equivalent I could find). As wack as this is, it seems maybe kinda necessary?